### PR TITLE
chore: Switch from edx-sphinx-theme to sphinx-book-theme

### PR DESCRIPTION
**Pull request by xitij2000**
The edx-sphinx theme is being deprecated, and replaced with sphinx-book-theme. This removes references to the deprecated theme and replaces them with the new standard theme for the platform.

**Testing instructions:** 
- Run `make docs` and see that the docs are generated properly and use the sphinx-book-theme

See https://github.com/openedx/edx-sphinx-theme/issues/184

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @xitij2000!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by mphilbrick211** [comment]
Hi @xitij2000 - looks like there's some failing tests on this one.

**Comment by xitij2000** [comment]
> Hi @xitij2000 - looks like there's some failing tests on this one.

It seems the test failures are happening due to a change in linting rules which were updated during the requirements upgrade. So I'm limiting all these PRs to just changes to docs related requirements.

**Comment by xitij2000** [comment]
The coverage failure seems to be due to reasons outside of this PR since this doesn't touch any non documentation code. 
Given that this repo is to be archived anyway, would it be OK to merge this as-is, if at all?

**Comment by mphilbrick211** [comment]
> The coverage failure seems to be due to reasons outside of this PR since this doesn't touch any non documentation code. Given that this repo is to be archived anyway, would it be OK to merge this as-is, if at all?

@jmbowman - flagging this for you. Thanks!

**Comment by Cup0fCoffee** [comment]
:+1:

- [x] I tested this: ran `make docs`
- [x] I read through the code

**Review by feanil — APPROVED** [review]
1 change, then I can merge.

**Comment by xitij2000** [comment]
@feanil Tests are failing in this and some other repos due to codecov being missing. 

I can remove the dependency but am not sure if this is the appropriate PR for it. Perhaps as a separate commit in the PRs? Or do I wait till that is corrected in a separate PR before rebasing these PRs?

**Comment by feanil** [comment]
I've got a separate PR to fix it, that I'm waiting for review on: https://github.com/openedx/bok-choy/pull/431

**Comment by feanil** [comment]
@xitij2000 you should be ableto rebase this now and it should pass tests.

**Comment by xitij2000** [comment]
> @xitij2000 you should be ableto rebase this now and it should pass tests.

Done!

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/bok-choy&entry.867055334=https://github.com/openedx/bok-choy/pull/426&entry.1484655318=https://github.com/xitij2000&entry.752974735=2023-03-24+07:30&entry.1917517419=2023-04-20+13:35&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
